engine/src/main/java/org/hibernate/search/bridge/MetadataProvidingFieldBridge.java

```diff
@@ -19,11 +19,11 @@
 public interface MetadataProvidingFieldBridge extends FieldBridge {
 
 	/**
-	 * Returns the names of the sortable fields created by this field bridge.
+	 * Allows this bridge to expose meta-data about the fields it creates.
 	 *
 	 * @param name The default field name; Should be used consistently with
 	 * {@link FieldBridge#set(String, Object, org.apache.lucene.document.Document, LuceneOptions)}.
-	 * @return The names of the sortable fields created by this field bridge. Never {@code null}.
+	 * @param builder Builder for exposing field-related meta-data
 	 */
-	Set<String> getSortableFieldNames(String name);
+	void configureFieldMetadata(String name, FieldMetadataBuilder builder);
```

Comment:
great idea to expose the builder!

Could you add a note in the javadoc about  the purpose of this contract?
As a know-nothing user, I don't think I'd understand what benefit I get from implementing this contract vs its parent.
Whilte at it, might be a good idea to warn about possible consequences of a mis-defined metadata?